shell/common/rasterizer.cc

```diff
@@ -265,9 +268,32 @@ RasterStatus Rasterizer::DoDraw(
   // TODO(liyuqian): in Fuchsia, the rasterization doesn't finish when
   // Rasterizer::DoDraw finishes. Future work is needed to adapt the timestamp
   // for Fuchsia to capture SceneUpdateContext::ExecutePaintTasks.
-  timing.Set(FrameTiming::kRasterFinish, fml::TimePoint::Now());
+  const auto raster_finish_time = fml::TimePoint::Now();
+  timing.Set(FrameTiming::kRasterFinish, raster_finish_time);
   delegate_.OnFrameRasterized(timing);
 
+  const fml::TimePoint latest_frame_target_time =
+      delegate_.GetLatestFrameTargetTime();
+  if (raster_finish_time > frame_target_time) {
+    const auto frame_budget_millis = delegate_.GetFrameBudget().count();
+    const auto frame_lag =
+        (latest_frame_target_time - frame_target_time).ToMilliseconds();
+    const int vsync_transitions_missed =
+        round(frame_lag) / round(frame_budget_millis);
```

Comment:
I'm not sure about the math here. Is this round() the standard math round? I don't think it helps this calculation.

The formula for a "rounded integer divide" is ((a + b/2) / b). In this case:
```
    const int vsync_transitions_missed = (frame_lag + frame_budget_millis/2) / frame_budget_millis;
```

Reply:
Yup, that is wrong. Not sure what I was thinking there. Let's say the frame budget is `16.666667`. I'd like us to list out `vsync_transitions_missed` for various values of `frame_lag`.

1. frame_lag = 10, missed = 1
2. frame_lag = 17, missed = 2
3. frame_lag = 20, missed = 2
4. frame_lag = 30, missed = 2
5. frame_lag = 34, missed = 3

Wouldn't regular integer integer division, without any rounding work there?

Reply:
Digging into the types used here a bit, it looks like some of them are millis expressed as doubles, which makes it easier. If that is the case, then `round(frame_lag / frame_budget_millis)` should be fine.

I think `frame_lag` is an int64_t, though which is not very accurate. Perhaps `ToMillisecondsF()` producing a double would be better? Then I think that both values are doubles and simply rounding (the math.h version) the result of their division would be sufficient.

Do my investigations check out?

Reply:
Yup. Amended my comment too :-)